metricbeat/module/kubernetes/util/metrics_cache.go

```diff
@@ -28,66 +29,74 @@ var PerfMetrics = NewPerfMetricsCache()
 const defaultTimeout = 120 * time.Second
 
 var now = time.Now
-var sleep = time.Sleep
+var after = time.After
```

Comment:
You are not serializing access to this global variable

Reply:
Yeah, there are some more things to improve around there.

I didn't want to touch this a lot, but at the end I also wanted the race detector to work and it wasn't with these global variables. I moved them to local variables in the caches.